### ISSUE TITLE
Loki Log Ingester per stream rate limit exceeded for prod logs as the number of nodes increases

### ISSUE THREAD

**Issue by computate** [nerc-project/operations#121]
[See here how the number of worker nodes in the OCP Prod cluster have doubled this month](https://grafana-open-cluster-management-observability.apps.nerc-ocp-infra.rc.fas.harvard.edu/explore?orgId=1&left=%5B%22now-30d%22,%22now%22,%22Observatorium%22,%7B%22exemplar%22:true,%22expr%22:%22count(kube_node_status_condition%7Bcondition%3D%5C%22Ready%5C%22,%20cluster%3D%5C%22nerc-ocp-prod%5C%22%7D)%20BY%20(cluster)%22%7D%5D). 
![image](https://user-images.githubusercontent.com/15114160/234044952-997b7e28-2bc6-4aaf-aa17-63bc3985137c.png)

See how this has caused the Loki Ingester pods in the Infra cluster to start rejecting logs sent from the Prod cluster with a Per stream rate limit exceeded error:
```bash
oc -n openshift-logging logs pod/logging-loki-ingester-1 | grep 'Per stream rate limit exceeded' | tail -n 1
```
reason: '**Per stream rate limit exceeded** (**limit: 3MB/sec**) while attempting to ingest for stream '{kubernetes_container_name=\"ocs-operator\", kubernetes_host=\"wrk-10\", kubernetes_namespace_name=\"openshift-storage\", kubernetes_pod_name=\"ocs-operator-cb6794cd9-hlggq\", log_type=\"infrastructure\"}' **totaling 112674B**, **consider splitting a stream via additional labels or contact your Loki administrator to see if the limit can be increased**'...

I don't see in the [Loki Operator spec](https://loki-operator.dev/docs/api.md/#loki-grafana-com-v1-IngestionLimitSpec) a way to configure the `Per stream rate limit` or splitting streams via additional labels, so I will reach out to the engineering team for ideas on how to get prod logs working again.

**Comment by joachimweyl** [comment]
RedHat [Case ticket](https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/auth?client_id=customer-portal&redirect_uri=https%3A%2F%2Faccess.redhat.com%2Fwebassets%2Favalon%2Fj%2Fincludes%2Fsession%2Fscribe%2F%3FredirectTo%3Dhttps%253A%252F%252Faccess.redhat.com%252Fsupport%252Fcases%252F%2523%252Fcase%252F03500822&state=0adf45d2-0e47-45de-8169-bc740457d30a&nonce=d12ca353-8dea-4506-8627-41abb82bc4be&response_mode=fragment&response_type=code&scope=openid) entered.

**Comment by joachimweyl** [comment]
@computate now that the PR is merged what are the next steps for this issue?

**Comment by computate** [comment]
@joachimweyl According to our Red Hat Support case, The code for configuring per stream rate limits "was merged into the OpenShift Logging 5.8 source release.  Since this is a change to the CRD, and therefore the API, it would normally not be something that can be backported to OpenShift Logging 5.7. Openshift Logging 5.8 is scheduled to be released in the latter part of this year."

The latest available version of the OpenShift Logging Operator is `5.7.3`, and the deployed version of the OpenShift Logging Operator in the NERC Prod infrastructure cluster is `5.6.5`. When a version `5.8` is available, there will be better default per stream rate limits, and we will be able to configure them higher.
